src/components/shapes/helpers.js

```diff
@@ -208,23 +224,22 @@ exports.getPathString = function(gd, options) {
         if(ya && ya.type === 'date') y2p = exports.decodeDate(y2p);
         return convertPath(options, x2p, y2p);
     }
-
     if(options.xsizemode === 'pixel') {
         var xAnchorPos = x2p(options.xanchor);
-        x0 = xAnchorPos + options.x0;
-        x1 = xAnchorPos + options.x1;
+        x0 = xAnchorPos + options.x0 + shiftUnitX * options.xshift;
+        x1 = xAnchorPos + options.x1 + shiftUnitX * options.xshift;
     } else {
-        x0 = x2p(options.x0);
-        x1 = x2p(options.x1);
+        x0 = x2p(options.x0) + shiftUnitX * options.xshift;
+        x1 = x2p(options.x1) + shiftUnitX * options.xshift;
```

Comment:
Many duplicates of `shiftUnitX * options.xshift` here.
Let's put it in a variable and use it in those places.

Reply:
done 👍